### PR TITLE
Feat/fly 2547

### PR DESCRIPTION
**Pull request by Hakob23**
## What

After claim, `registerClaimedPegOut` mirrors the escrow-built `Quotes.PegOutQuote` into `_pegOutQuotes[requestHash]` (same store as `depositPegOut`), so today’s `_validatePegOutTransaction` accepts claim-fed records with no validation-path changes. Escrow-only access control and LP `SignatureValidator.verify` are unchanged. Public `depositPegOut` stays for the dual path until it is removed in a dedicated PR.

## Why

Claim already forwarded RBTC and registry timing, but never wrote the quote body, so settlement hit `QuoteNotFound`. Without one mirrored record feeding the existing validator, refund/SPV validation cannot run on commit-first peg-outs (drift risk of a second quote/validation path).

## Type of Change

- [ ] Bug fix (non-breaking change)
- [x] New feature (non-breaking change)
- [ ] Breaking change
- [ ] Refactor (no intended behavior change)
- [ ] Security hardening
- [ ] Tests only
- [ ] Build/CI/Tooling
- [ ] Documentation

## Affected Areas

- [ ] PegIn flow (`registerPegIn`, `callForUser`, quote validation)
- [x] PegOut flow (`depositPegout`, `refundPegOut`, `refundUserPegOut`)
- [ ] Liquidity Provider lifecycle (register/update/status/resign/collateral/withdraw)
- [ ] Pause/operational controls
- [x] Quote hashing/signature logic
- [x] Bitcoin tx / PMT / merkle validation
- [ ] Deployment or upgrade scripts / Makefile targets
- [ ] CI workflows (`ci`, `fuzz`, `formal`, `slither`, `codeql`)
- [ ] Docs

## Risk & Security Review

- [x] Access control and authorization paths reviewed
- [x] Reentrancy and external call paths reviewed
- [x] Storage layout / upgrade safety reviewed (if upgradeable code changed)
- [x] Time/block/confirmation assumptions reviewed
- [x] BTC tx output/index assumptions reviewed (if pegout validation touched)
- [x] No secrets or sensitive values added

## Test Plan

List what you ran locally and the outcome:

- [ ] `npm run lint:sol`
- [ ] `npm run lint:ts`
- [ ] `npm run compile`
- [x] `npm test` — `forge test --match-path 'test/pegout-escrow/**'` (35 passed, including `test_S13_1_ClaimFedRecord_PassesValidatePegout`); `DeployPegOutEscrow` suite green
- [ ] `npm run test:fuzz` (if logic/state transitions changed)
- [ ] `npm run test:invariant` (if invariant-sensitive logic changed)
- [ ] `npm run test:formal` (if formal-verification-sensitive logic changed)
- [ ] `npm run test:integration` (if integration paths changed)
- [ ] `npm run test:e2e` (if deployment/ownership flows changed)

Additional notes on test coverage, edge cases, and any tests intentionally skipped:

- Acceptance: claim → FFI BTC tx (`OP_RETURN` = `requestHash`) → `validatePegout` returns mirrored quote fields.
- `_validatePegOutTransaction` unmodified; only field *storage* at register time changed.
- Non-escrow `registerClaimedPegOut` still covered by existing deploy tests (`OnlyPegOutEscrow`).
- Out of scope: third-party refund / `FULFILLED` notify, short-delivery floor, regtest e2e `refundPegOut`.
- TODO in escrow suite: rename ticket-tagged test names once review no longer needs them.

## Deployment & Ops Impact

- [x] No deployment impact
- [ ] Requires deployment
- [ ] Requires upgrade
- [ ] Env/config changes required (`.env`, network RPCs, verification, etc.)
- [ ] Runbook/update instructions added to PR description

If deployment/upgrade is needed, include target network(s), simulation command(s), and rollback considerations.

## Related Issues

- Jira: [FLY-2547](https://rsklabs.atlassian.net/browse/FLY-2547)

## Reviewer Notes

- Focus on `registerClaimedPegOut`: after sig verify it now does `_pegOutQuotes[requestHash] = quote` alongside registry writes — keyed by escrow `requestHash` (OP_RETURN id), not `keccak256(encodePegOutQuote)`.
- One struct (`Quotes.PegOutQuote`), one validation path; reconstruction stays in escrow `_buildQuote` only.
- Quote body is intentionally duplicated in storage (escrow + PegOutContract) so settlement internals stay unchanged; not a second builder/validator.
- `depositPegOut` not retired/gated (legacy quote path kept for existing LPS/SDK).

### CODE REVIEW

File: src/PegOutContract.sol [hunk 1]
```diff
@@ -188,6 +188,9 @@ contract PegOutContract is
             revert SignatureValidator.IncorrectSignature(quote.lpRskAddress, eip712Hash, signature);
         }
 
+        // Mirror depositPegOut: same Quotes.PegOutQuote + registry so _validatePegOutTransaction
```
Comment (Luisfc68) [inline src/PegOutContract.sol:191]: Again, this is not needed if you requse the full depositPegOut here with the extra checks, this might not be the last issue generated by this

**Review by Luisfc68 — CHANGES_REQUESTED** [review]

File: src/PegOutContract.sol (hunk 1 above)
Comment (Luisfc68) [reply src/PegOutContract.sol:191]: I really recommend doing that change I think is safer for several aspects, it uses an existing funcion, it keeps the internal logic of the old contract unchanged, it separates better the responsibilities

Comment (Hakob23) [reply src/PegOutContract.sol]: Agreed — claim goes through `depositPegOut` (no parallel register path). https://github.com/rsksmart/liquidity-bridge-contract/pull/549
